### PR TITLE
builds(deps): use miniupnpc package instead of submodule

### PR DESCRIPTION
**Pull request by ReenigneArcher**
## Description

Use package of miniupnpc instead of submodule.

Based on discussion here: https://github.com/LizardByte/Sunshine/pull/1969

See also: #1959 

Todo:
- [x] Flatpak
- [x] Fix Windows link error

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Dependency update (updates to dependencies)
- [x] Documentation update (changes to documentation)
- [x] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I want maintainers to keep my branch updated

### CODE REVIEW

**Comment by ReenigneArcher** [comment]
@chewi just FYI

File: cmake/dependencies/common.cmake [hunk 1]
```diff
@@ -16,26 +16,16 @@ find_package(Threads REQUIRED)
...
-    include_directories(SYSTEM third-party/miniupnp/miniupnpc/include)
-endif()
+# Use includedir pkg-config variable rather than MINIUPNP_INCLUDE_DIRS
+# defined above. The latter may be blank, as pkg-config sometimes omits -I
+# flags for directories that are searched by default (e.g. /usr/include),
+# but we need a value to append /miniupnpc to. Normally source files would
+# prefix their #include directives with miniupnpc/, but this does not align
+# with the directory structure of the git submodule used below.
+pkg_get_variable(MINIUPNP_INCLUDE_DIR miniupnpc includedir)
+include_directories(SYSTEM ${MINIUPNP_INCLUDE_DIR}/miniupnpc)
```
Comment (chewi) [inline cmake/dependencies/common.cmake:28]: Now you can get rid of this kludge and just do this instead:
```cmake
include_directories(SYSTEM ${MINIUPNP_INCLUDE_DIRS})
```

File: cmake/dependencies/common.cmake [hunk 2]
```diff
@@ -16,26 +16,9 @@ find_package(Threads REQUIRED)
...
-    set(UPNPC_BUILD_SHARED OFF CACHE BOOL "No shared libraries")
-    set(UPNPC_BUILD_TESTS OFF CACHE BOOL "Don't build tests for miniupnpc")
-    set(UPNPC_BUILD_SAMPLE OFF CACHE BOOL "Don't build samples for miniupnpc")
-    set(UPNPC_NO_INSTALL ON CACHE BOOL "Don't install any libraries build for miniupnpc")
-    set(MINIUPNP_LIBRARIES libminiupnpc-static)
-    add_subdirectory(third-party/miniupnp/miniupnpc)
-    include_directories(SYSTEM third-party/miniupnp/miniupnpc/include)
-endif()
+pkg_check_modules(MINIUPNP miniupnpc REQUIRED)
+pkg_get_variable(MINIUPNP_INCLUDE_DIR miniupnpc includedir)
```
Comment (chewi) [inline cmake/dependencies/common.cmake:20]: To be clear, this `pkg_get_variable` line is not needed any more.

Comment (ReenigneArcher) [reply cmake/dependencies/common.cmake:20]: Got it! Thanks!

**Comment by ReenigneArcher** [comment]
Just need to resolve the Windows linking error and this should be good to go.

File: src/upnp.cpp [hunk 3]
```diff
@@ -2,8 +2,9 @@
  * @file src/upnp.cpp
  * @brief todo
  */
-#include <miniupnpc.h>
-#include <upnpcommands.h>
+#define MINIUPNP_STATICLIB
```
Comment (chewi) [inline src/upnp.cpp:5]: I can see that only affects Windows, but I'd feel better if it was behind an `#ifdef _WIN32`.

Comment (ReenigneArcher) [reply src/upnp.cpp]: Moved to cmake.

**Pull request merged**